nifi-nar-bundles/nifi-provenance-repository-bundle/nifi-persistent-provenance-repository/src/test/groovy/org/apache/nifi/provenance/EncryptedSchemaRecordReaderWriterTest.groovy

```diff
@@ -273,10 +273,10 @@ class EncryptedSchemaRecordReaderWriterTest extends AbstractTestRecordReaderWrit
         RecordReader incompatibleReader = new EventIdFirstSchemaRecordReader(efis, encryptedJournalFile.getName(), incompatibleTocReader, MAX_ATTRIBUTE_SIZE)
         logger.info("Generated standard reader (attempting to read encrypted file): ${incompatibleReader}")
 
-        def msg = shouldFail(EOFException) {
+        def msg = shouldFail(IOException) {
             ProvenanceEventRecord encryptedEvent = incompatibleReader.nextRecord()
         }
         logger.expected(msg)
-        assert msg =~ "EOFException: Failed to read field"
+        assert msg =~ "IOException: Invalid Boolean value found when reading 'Repetition' of field"
```

Comment:
This change is not clear to me. The behavior of the test seems to have changed here. The test was testing for a very specific failure. This appears to have changed the failure that the test expects...

Reply:
I spent some time trying to understand this unit test and what it was asserting: that standard provenance record readers can't read encrypted provenance events. 

It seems that the proof that a standard record reader's inability to read encrypted events was just confirming that it choked when trying to deserialize the encrypted events. For example, the standard reader doesn't know it's encrypted; it just chokes when it can't deserialize the event as it expects. As far as i can tell, since the serialization changed (e.g. the serialized event looks different), the reader chokes at a different spot. Hence the change in exception that's thrown. Does that make sense?